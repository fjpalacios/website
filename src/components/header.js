import * as React from 'react'
import { useI18next } from 'gatsby-plugin-react-i18next'
import contactEn from '../../content/en/contact.js'
import contactEs from '../../content/es/contact.js'
import './header.scss'

const Header = () => {
  const { t, language } = useI18next()
  const contact = language === 'en' ? contactEn : contactEs

  return (
    <header className="header">
      <h1 className="header__title">{t('title')}</h1>
      <h2 className="header__subtitle">{t('subtitle')}</h2>
      <div className="header__contact">
        {contact.map((item, key) => {
          return (
            <div key={key} className={`header__contact__item ${item.icon}`}>
              <a href={item.link} target="_blank" rel="noreferrer">
                <i className={`icon-${item.icon}`}></i>
                <p>{item.text}</p>
              </a>
            </div>
          )
        })}
      </div>
    </header>
  )
}

export default Header
