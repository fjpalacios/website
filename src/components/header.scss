.header {
  display: grid;
  grid-column: span 12;
  grid-template-areas:
    'title'
    'subtitle'
    'contact';
  margin-bottom: 40px;
  padding: 20px;
  transition: all .45s ease;
  background-color: $primary;
  color: $inverted-text;

  @include medium-and-up {
    grid-template-areas:
      'title contact'
      'subtitle contact';
  }

  @include print {
    grid-template-areas:
      'title contact'
      'subtitle contact';
    margin-bottom: 20px;
    padding: 10px;
  }

  &__title {
    grid-area: title;
    margin: 0;
    font-size: 1.2em;
    font-weight: bold;
    text-transform: uppercase;

    @include small-and-up {
      font-size: 2em;
    }

    @include print {
      font-size: 1.7em;
    }
  }

  &__subtitle {
    grid-area: subtitle;
    margin: 5px 0 0;
    font-size: 1em;
    font-weight: 100;
    text-transform: uppercase;

    @include small-and-up {
      margin: 0;
      font-size: 1.5em;
    }

    @include print {
      margin: 0;
      font-size: 1.2em;
    }
  }

  &__contact {
    display: grid;
    grid-area: contact;
    grid-template-columns: repeat(auto-fit, 20px);
    justify-content: space-between;
    margin-top: 20px;
    font-size: .8em;

    @include medium-and-up {
      grid-template-columns: repeat(2, 150px);
      justify-content: end;
      margin-top: 0;
    }

    @include print {
      grid-template-columns: repeat(2, 110px);
      justify-content: end;
      margin-top: 0;
    }

    &__item {
      cursor: pointer;

      a {
        display: flex;
        transition: all .45s ease;
        color: $header-link;

        i {
          margin-right: .5em;
          font-size: 1.5em;

          @include medium-and-up {
            font-size: 1em;
          }

          @include print {
            font-size: 1em;
          }
        }

        p {
          display: none;

          @include medium-and-up {
            display: block;
            margin: 0;
            padding: 0;
          }

          @include print {
            display: block;
            margin: 0;
            padding: 0;
          }
        }
      }
    }
  }
}

@include screen {
  .phone {
    display: none;
  }

  .globe {
    display: none;
  }
}

@include print {
  .twitter {
    display: none;
  }
}
