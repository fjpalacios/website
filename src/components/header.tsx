import './header.scss'
import React, { FunctionComponent, ReactElement } from 'react'
import { useI18next, useTranslation } from 'gatsby-plugin-react-i18next'
import contactEn from '../../content/static/contact/en'
import contactEs from '../../content/static/contact/es'

export const Header: FunctionComponent = (): ReactElement => {
  const { t } = useTranslation()
  const { defaultLanguage, language } = useI18next()
  const contact = language === defaultLanguage ? contactEn : contactEs

  return (
    <header className="header">
      <h1 className="header__title">{t('title')}</h1>
      <div className="header__subtitle">{t('subtitle')}</div>
      <div className="header__contact">
        {contact.map((item, key) => {
          return (
            <div key={key} className={`header__contact__item ${item.icon}`}>
              <a href={item.link} target="_blank" rel="noreferer">
                <i className={`icon-${item.icon}`}></i>
                <p>{item.text}</p>
              </a>
            </div>
          )
        })}
      </div>
    </header>
  )
}
