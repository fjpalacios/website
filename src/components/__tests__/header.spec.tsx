import createComponentWithIntl from '../../../__helpers__/i18n-component'
import { Header } from '../header'
import React from 'react'

describe('Header', () => {
  it('should render properly', () => {
    const header = createComponentWithIntl(<Header />)
    expect(header).toMatchSnapshot()
  })

  it('should show English language by default', () => {
    const header = createComponentWithIntl(<Header />)
    const subTitle = header.find('.header__subtitle').text()
    expect(subTitle).toBe('Software Developer')
  })

  it('should show Spanish language if required', () => {
    const header = createComponentWithIntl(<Header />, 'es')
    const subTitle = header.find('.header__subtitle').text()
    expect(subTitle).toBe('Desarrollador de software')
  })
})
