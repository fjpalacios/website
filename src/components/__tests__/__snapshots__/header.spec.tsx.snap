// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "about": Object {
        "education": "Education",
        "experience": "Experience",
        "freelance": "Freelance work",
        "me": "About me",
        "projects": "Side projects",
        "talks": "Talks / workshops",
        "volunteering": "Volunteering",
      },
      "menu": Object {
        "switchTheme": "Switch Theme",
        "toEnglish": "Switch to English language",
        "toSpanish": "Switch to Spanish language",
      },
      "metaData": Object {
        "description": "Francisco Javier Palacios Pérez's personal website, Software Developer based in Valencia (Spain)",
        "image": "https://fjp.es/logo.jpg",
        "twitterUsername": "@fjpalacios",
        "url": "https://fjp.es",
      },
      "subtitle": "Software Developer",
      "title": "Francisco Javier Palacios Pérez",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <Header>
    <header
      className="header"
    >
      <h1
        className="header__title"
      >
        Francisco Javier Palacios Pérez
      </h1>
      <h2
        className="header__subtitle"
      >
        Software Developer
      </h2>
      <div
        className="header__contact"
      >
        <div
          className="header__contact__item linkedin"
          key="0"
        >
          <a
            href="https://www.linkedin.com/in/fjpalacios/"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-linkedin"
            />
            <p>
              fjpalacios
            </p>
          </a>
        </div>
        <div
          className="header__contact__item mail"
          key="1"
        >
          <a
            href="mailto:javi@fjp.es"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-mail"
            />
            <p>
              javi@fjp.es
            </p>
          </a>
        </div>
        <div
          className="header__contact__item github-circled"
          key="2"
        >
          <a
            href="https://github.com/fjpalacios"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-github-circled"
            />
            <p>
              fjpalacios
            </p>
          </a>
        </div>
        <div
          className="header__contact__item globe"
          key="3"
        >
          <a
            href="https://fjp.es"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-globe"
            />
            <p>
              fjp.es
            </p>
          </a>
        </div>
        <div
          className="header__contact__item location"
          key="4"
        >
          <a
            href="https://osm.org/go/b_qODpV--"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-location"
            />
            <p>
              Valencia, ES
            </p>
          </a>
        </div>
        <div
          className="header__contact__item phone"
          key="5"
        >
          <a
            href="tel:0034671995762"
            rel="noreferer"
            target="_blank"
          >
            <i
              className="icon-phone"
            />
            <p>
              671995762
            </p>
          </a>
        </div>
      </div>
    </header>
  </Header>
</IntlProvider>
`;
